### ISSUE TITLE
Clang emits false positive warning '::' and '*' tokens forming pointer to member type appear in different macro expansion contexts

### ISSUE THREAD

**Issue by ranaanoop** [llvm/llvm-project#92259]
The following valid/well-formed program produces a false positive warning in clang:

```
#define FOO A::

struct A
{
    int x;
};

int FOO *ptr = &A::x;
```
Clang says:

```
<source>:8:5: warning: '::' and '*' tokens forming pointer to member type appear in different macro expansion contexts [-Wcompound-token-split-by-macro]
```

**Comment by MitalAshok** [comment]
Clang considers `::*` to be a "compound token"

https://github.com/llvm/llvm-project/blob/61da6366d043792d7db280ce9edd2db62516e0e8/clang/include/clang/Parse/Parser.h#L1143-L1144

https://github.com/llvm/llvm-project/blob/61da6366d043792d7db280ce9edd2db62516e0e8/clang/test/Parser/compound-token-split.cpp#L41

In the standard, these two tokens always appears written `::*` without a space, but the definition doesn't disallow a space: https://wg21.link/dcl.mptr#2

It's not a false positive, since that warning is looking exactly for a compound token split across macro boundaries.

For reference, we also warn on:

```c++
#define LSQUARE [
[ LSQUARE noreturn ] ]
int f();
```
